stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt

```diff
@@ -116,9 +116,10 @@ public fun ChatClient.watchChannelAsState(
     cid: String,
     messageLimit: Int,
     coroutineScope: CoroutineScope = CoroutineScope(DispatcherProvider.IO),
+    skipMessages: Boolean = false
```

Comment:
Also, we are sending `skipMessages = true` within the `MessageComposerViewModel` and `MessageHeaderViewModel` but debouncing is done using checking if two `QueryChannelRequest`s are the same. I updated it's equals function to take into account `skipMessages`.

Reply:
What if instead of introducing an additional parameter we can manipulate with `messageLimit`. We could expose the limit via constructor in each view model and set it to `0` if `messageId != 0`. What do you think?

Reply:
Hmm, in stead of exposing `messageLimit` maybe we can expose `messageId` and then set the limit to 0 if we find a non-null `messageId`. Introducing `messageLimit` to composer and header might be confusing and in the case of list would mean that we take it as param for pagination and loading of data. How does this sound?

Reply:
Makes sense

Reply:
Maybe we don't have to pass `messageId` to all of the VMs. The only VM that is using the message list is the `MessageListViewModel`. We can pass `messageLimit = 0` to all others vms and let them refresh channel statate without messages. Then we can just do the check inside `MessageListViewModel`. The message list query won't be debounced due to different `messageLimit`.

Reply:
I think we can just pass in the messageId to the ListViewModel as the "observed/focused message". 
----

It might be useful to pass in the messageId to all VMs in the future, e.g. if someone wants to outline in the header which message is being observed, or alternatively in the Composer which message is being jumped to and edited.

But these seem like custom use cases and will most likely be confusing to our customers if it doesn't exist as default behavior they can relate to. Hence why I propose to just add it where it makes sense now.

Reply:
@adasiewiczr refactored so we use `messageLimit` to control upserting of messages on itial loads.